### DISCUSSION TITLE
How to setup tcp stream listener router

### DISCUSSION THREAD

**Discussion by naolaregaqena** [Kong/kong#10162] [Help] [answered]
I was trying to proxy rabbitmq through kong, and i could find clear explanation about it.
I have enabled `stream_listen` as so.
```
# nginx section
stream_listen = *:7000 reuseport
```

**Kong gateway oss**: `2.8.1`
**Postgres**: `9.6-alpine`
**Docker**: `20.10.21`

**Comment by chronolaw** [comment, marked as answer]
You should define a tcp route entity(https://docs.konghq.com/gateway/latest/admin-api/#route-object).

**Reply by gszr** [reply]
As Chrono mentions, setting the `stream_listen` only defines `addresses:port` pairs Kong will listen on. As mentioned in https://docs.konghq.com/gateway/latest/how-kong-works/routing-traffic/#proxy-tcptls-traffic: 
```
To use this mode, aside from defining stream_listen, appropriate route/service object with protocol types of tcp or tls should be created.
```
